config/crd/bases/exp.cluster.x-k8s.io_machinepools.yaml

```diff
@@ -77,6 +77,13 @@ spec:
                   rollingUpdate:
                     description: Rolling update config params. Present only if MachineDeploymentStrategyType = RollingUpdate.
                     properties:
+                      deletePolicy:
+                        description: DeletePolicy defines the policy used by the MachineSet to identify nodes to delete when downscaling. Valid values are "Random, "Newest", "Oldest" When no value is supplied, the default DeletePolicy of MachineSet is used
+                        enum:
+                        - Random
+                        - Newest
+                        - Oldest
+                        type: string
```

Comment:
This looks like a potential unintended consequence, is there any change to the MachinePool type definition to avoid exposing this?

Reply:
MachinePoolSpec has reference to  DeletePolicy through `MachineDeploymentStrategy.RollingUpdate`. Not sure if there's a way to avoid this.

Reply:
@CecileRobertMichon do you have any thoughts here?

Reply:
It could apply to MachinePools, but it might also put undue constrains on the provider implementation. Because of the previous statement, I think reusing `MachineDeploymentSpec` will be problematic. Some implementors of `MachinePool` may not be able to replicate the specific behavior of a MachineDeployment.

Reply:
Should I create an issue for removing `MachineDeploymentStrategy` from `MachinePoolSpec`?

Reply:
Yes please. You can assign me.

Reply:
Created #3813 to address this.